python/GafferImageTest/ResampleTest.py

```diff
@@ -0,0 +1,146 @@
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import os
+import shutil
+import unittest
+import subprocess
+
+import IECore
+
+import Gaffer
+import GafferTest
+import GafferImage
+
+class ResampleTest( GafferTest.TestCase ) :
+
+	def testDataWindow( self ) :
+
+		r = GafferImage.Resample()
+		r["dataWindow"].setValue(
+			IECore.Box2f(
+				IECore.V2f( 10.5, 11.5 ),
+				IECore.V2f( 20.5, 21.5 )
+			)
+		)
+
+		self.assertEqual(
+			r["out"]["dataWindow"].getValue(),
+			IECore.Box2i(
+				IECore.V2i( 10, 11 ),
+				IECore.V2i( 21, 22 )
+			)
+		)
+
+	def testExpectedOutput( self ) :
+
+		def __test( fileName, size, filter ) :
+
+			inputFileName = os.path.dirname( __file__ ) + "/images/" + fileName
+
+			reader = GafferImage.ImageReader()
+			reader["fileName"].setValue( inputFileName )
+
+			resample = GafferImage.Resample()
+			resample["in"].setInput( reader["out"] )
+			## \todo Adjust for #1438
+			resample["dataWindow"].setValue( IECore.Box2f( IECore.V2f( 0 ), IECore.V2f( size.x, size.y ) - IECore.V2f( 1 ) ) )
+			resample["filter"].setValue( filter )
+			resample["boundingMode"].setValue( GafferImage.Sampler.BoundingMode.Clamp )
+
+			crop = GafferImage.Crop()
+			crop["in"].setInput( resample["out"] )
+			crop["area"].setValue( IECore.Box2i( IECore.V2i( 0 ), size ) )
+
+			outputFileName = "/tmp/gafferImageResampleTest/%s_%dx%d_%s.exr" % ( os.path.splitext( fileName )[0], size.x, size.y, filter )
+			writer = GafferImage.ImageWriter()
+			writer["in"].setInput( crop["out"] )
+			writer["fileName"].setValue( outputFileName )
+			writer.execute()
+
+			result = GafferImage.ImageReader()
+			result["fileName"].setValue( writer["fileName"].getValue() )
+
+			expected = GafferImage.ImageReader()
+			expected["fileName"].setValue(
+				"%s/images/%s_%dx%d_%s.exr" % (
+					os.path.dirname( __file__ ),
+					os.path.splitext( fileName )[0],
+					size.x,
+					size.y,
+					filter
+				)
+			)
+
+			self.assertFalse(
+				IECore.ImageDiffOp()(
+					imageA = expected["out"].image(),
+					imageB = result["out"].image()
+				).value
+			)
+
+			if False : # Enable to write out images for visual comparison with OIIO
```

Comment:
If this is enabled, the files will be created, but then immediately deleted during the tearDown part of the test